### ISSUE TITLE
Pen-highlighter combined tool for annotation during presentation

### ISSUE THREAD

**Issue by arechnitzer** [xournalpp/xournalpp#2192]
Thanks for xournal++ - It is going to be very useful this term. I have a feature request.. and/or looking for guidance as to how this could be done with a plugin?

I am using xournal++ to annotate PDF slides while I present them and recording the result. My slides will have a black background and white text and I'm going to annotate them with pink and blue. It would be helpful to make these annotations stand out more from the background if I could get a white highlight around the annotation.  To be more precise... 

* user draws stroke S1 using pen-tool
* plugin (or similar) copies S1 to new stroke S2, then changes colour of S2 to white and makes it thicker
* plugin (or similar) pastes S2 under S1

or

* user draws stroke S1 using pen-tool
* plugin copies S2 under S1 using the highlighter tool

**Comment by rolandlo** [comment]
My first impression is that this could be hacked up, but doing it as a plugin would make it quite a bit harder, since the plugin would have to communicate with the C++ code a lot. In view of the fact, that this feature request is quite special, it would probably end up as your personal modified version of Xournal++, without a real chance to get into the master branch. As long as you don't want to update with the other changes pushed to the master branch, this may be OK. Otherwise you will sooner or later run into merge conflicts.
One thing is unclear to me in your description: Should the stroke S2 be created only after stroke S1 is finished? Or should it be updated all the time while a stroke is drawn?

**Comment by arechnitzer** [comment]
Created after would perhaps be simplest?

So the plugin interface can't do things like "copy object" "change colour" etc?

**Comment by rolandlo** [comment]
Yes, "created after" would be simpler. 
The plugin interface would have to be extended for that. It doesn't have much functionality so far. The even bigger problem is, that the plugin would have to be notified whenever a stroke has been completed.

**Comment by arechnitzer** [comment]
Oh well. "Thicker pen" sounds like a much simpler solution to this issue.

**Comment by rolandlo** [comment]
That's true. Have you noticed, that you can make the pen size "very thick" from the menu `tools->pen options`?
Moreover changing the color(s) (making the lighter) might also help.

**Comment by arechnitzer** [comment]
Yup - thats exactly what I'm doing.
(also covering part of the slide using mostly-opaque black filled black rectangles which I can then delete to reveal content)
